components/controlled_dialog.tsx

```diff
@@ -0,0 +1,99 @@
+import * as React from "react";
+
+import { Button, IButtonProps, Intent } from "@blueprintjs/core";
+import { CardActions } from "df/components/card";
+import { Dialog } from "df/components/dialog";
+
+export interface IProps {
+  title?: string;
+  ref?: React.RefObject<ControlledDialog>;
+  button?: React.ReactElement<any>;
+  onConfirm?: () => void;
+  onClose?: (e: any) => void;
+  confirmButtonProps?: IButtonProps & React.ButtonHTMLAttributes<HTMLButtonElement>;
```

Comment:
If we are moving this here, let's maybe clean this up? All this props for the button is not very flexible, could replace with a `confirmButton` prop and let the user pass in a component when overriding. could do same for `cancelButton`

Reply:
I decided against this in the end. Practically every component that uses ControlledDialog wants to change at least one prop, and I don't like the idea of them all changing to pass in a whole button